### PR TITLE
Added feature/slider-draft

### PR DESCRIPTION
**Pull request by kjec2f**
**Describe pull-request**  
_Describe what the pull-request is about_
  Added slider-draft
**Solving issue**  
_Add which issue this pull-request solves by adding # plus the number of the issue (for example #123)_
Fixes: #

**How to test**  
_Add description how to test if possible_
1. Go to storybook > component > slider
2. Desgin similar to "https://www.figma.com/proto/x0Lak62QgvbOuXW8bZFb9D/slider?page-id=80%3A379&node-id=80%3A578&viewport=677%2C691%2C0.11236094683408737&scaling=min-zoom"

**Screenshots**  
_If applicable, add screenshots to help explain_

**Additional context**  
_Add any other context about the pull-request here._

### CODE REVIEW

**Comment by mmexvr** [comment]
conflict

**Comment by mmexvr** [comment]
I think there is an important prop missing, so if you have a range it should be possible to change the value it should be between. So if you have a price range for a product you might need to have bigger or smaller numbers.

So it should be possible to change the `min` and `max` attribute on the html to your need, because it might be needed for developers to change that attribute and use it. You might have a range between 2000-3000 and not 0-100 all the time.

Same with `value` attribute for the input element is important that they can access and use, they it should be able to set a default start value or not

`<input type="range" min="1" max="100" value="50" class="slider" id="myRange">`

File: components/src/components/slider/slider.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        ></input>
+        <div
+          class="range-slider"
+          style={{
+            '--minval': `${this.rangeStyle['--val']}`,
+            '--maxval': `${this.secondRangeStyle['--val']}`
+          }}
+        >
+          <input
+            min="1"
```
Comment (mmexvr) [inline components/src/components/slider/slider.tsx:185]: Should be able to customise this value

Comment (mmexvr) [reply components/src/components/slider/slider.tsx:185]: Same with the other min and max on other examples

File: components/src/components/slider/slider.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+          ></input>
+        </div>
+        <input
+          min="1"
+          max="100"
+          type="text"
+          value={`${this.secondRangeStyle['--val']}`}
+          ref={(el) => (this.rightInputTextRef = el as HTMLInputElement)}
+          class="input-text"
+          onKeyUp={(ev) => this.handleOnChangeRight(ev)}
```
Comment (mmexvr) [inline components/src/components/slider/slider.tsx:210]: Can we use just on change and not keyup, or is there a specific reason for keyup?

File: components/src/components/slider/slider.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  leftInputTextRef!: HTMLInputElement;
+  rightInputTextRef!: HTMLInputElement;
+
+  @State() leftRangeOldValue: string;
+  @State() rightRangeOldValue: string;
+
+  @State() rangeStyle = {
+    '--min': '1',
+    '--max': '100',
+    '--val': '20'
```
Comment (mmexvr) [inline components/src/components/slider/slider.tsx:25]: value should be default 0 and be customisable for the user, what the default value should start on

**Review by mmexvr — CHANGES_REQUESTED** [review]
Above comments :)

**Comment by alexlioliopoulos** [comment]
We are also missing the hover state in the component.

**Comment by mmexvr** [comment]
Can you seperate the examples so it has its own page in storybook, so duel point has one menu for example

File: components/src/components/slider/slider.tsx (hunk 2 above)
Comment (kjec2f) [reply components/src/components/slider/slider.tsx:210]: In order to detect the value and show error if the value is not correct before the user release

**Comment by alexlioliopoulos** [comment]
From my point of view, great work!

**Comment by mmexvr** [comment]
@alexlioliopoulos Should the slider be the size of where it is placed or is it always the same small size, can it be for example 6 col big?

**Comment by alexlioliopoulos** [comment]
> @alexlioliopoulos Should the slider be the size of where it is placed or is it always the same size, 100px in this case, or can it be for example 6 col big instead?
> 
> I would assume there are cases where you want a bigger slider?

Great point! I suppose you are right, so we should give devs the option to change the width of it (but not the height).

**Comment by mmexvr** [comment]
> > @alexlioliopoulos Should the slider be the size of where it is placed or is it always the same size, 100px in this case, or can it be for example 6 col big instead?
> > I would assume there are cases where you want a bigger slider?
> 
> Great point! I suppose you are right, so we should give devs the option to change the width of it (but not the height).

Yes! So instead of having a set width on the slider, remove that and let it grow with the parent element instead. So if the parent is 200px width, then the slider will be that.

**Review by mmexvr — APPROVED** [review]

**Comment by mmexvr** [comment]
Looks good! 

Some options to add in the future:
- Add step option
- How the state should be when they are above each other

**Pull request merged**